lib/galaxy/model/migrate/versions/0152_created_from_basename.py

```diff
@@ -0,0 +1,46 @@
+"""
+Adds created_from_basename to dataset.
+"""
+import datetime
+import logging
+
+from sqlalchemy import Column, MetaData, Table, TEXT
+
+now = datetime.datetime.utcnow
+log = logging.getLogger(__name__)
+metadata = MetaData()
+
+
+def upgrade(migrate_engine):
+    metadata.bind = migrate_engine
+    print __doc__
+    metadata.reflect()
+
+    created_from_basename_column = Column("created_from_basename", TEXT, default=None, )
+
+    __add_column(created_from_basename_column, "dataset", metadata)
+
+
+def downgrade(migrate_engine):
+    metadata.bind = migrate_engine
+    metadata.reflect()
+
+    __drop_column("created_from_basename", "dataset", metadata)
+
+
+def __add_column(column, table_name, metadata, **kwds):
```

Comment:
Probably want to use the new utilities in galaxy.model.migrate.versions.util -- add_column/drop_column?

Reply:
Hadn’t realized that had merged, sounds great. I’ll redo this and give those a shot.